src/machine/machinectl.c

```diff
@@ -487,14 +533,21 @@ static int print_addresses(sd_bus *bus, const char *name, int ifi, const char *p
         if (r < 0)
                 return r;
 
+        addresses = strdup(prefix);
+        if (!addresses)
+                return log_oom();
+        prefix = "";
+
         r = sd_bus_message_enter_container(reply, 'a', "(iay)");
         if (r < 0)
                 return bus_log_parse_error(r);
 
         while ((r = sd_bus_message_enter_container(reply, 'r', "iay")) > 0) {
                 int family;
                 const void *a;
+                char *tmp;
                 size_t sz;
+                char buf_ifi[24];
```

Comment:
why 24? Please determine the exact size needed with `DECIMAL_STR_MAX(int)` instead here.